pkg/routeagent_driver/handlers/healthchecker/healthchecker_test.go

```diff
@@ -0,0 +1,409 @@
+				t.pingerMap[healthCheckIP1].AwaitStart()
+
+				endpoint1.Spec.Hostname = "newHostName"
+				t.UpdateEndpoint(endpoint1)
+
+				pingerObject, found := t.pingerMap[endpoint1.Spec.HealthCheckIP]
+				Expect(found).To(BeTrue())
+				Expect(pingerObject.GetIP()).To(Equal(healthCheckIP1))
+
+				t.pingerMap[healthCheckIP1].AwaitNoStop()
+			})
+		})
+	})
+
+	When("a remote Endpoint is deleted", func() {
+		t := newTestDriver(true)
+		It("should stop the pinger", func() {
+			endpoint1 := t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
+			t.pingerMap[healthCheckIP1].AwaitStart()
+
+			t.DeleteEndpoint(endpoint1.Name)
+			t.pingerMap[healthCheckIP1].AwaitStop()
+		})
+	})
+})
+
+var _ = Describe("Transition of nodes", func() {
+	When("a node transition to gateway node", func() {
+		t := newTestDriver(true)
+		It("should stop the pinger", func() {
+			_ = t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
+			t.pingerMap[healthCheckIP1].AwaitStart()
+
+			_ = t.CreateLocalHostEndpoint()
+			t.pingerMap[healthCheckIP1].AwaitStop()
+		})
+	})
+
+	When("a node transition to non-gateway node", func() {
+		t := newTestDriver(true)
+		It("should start the pinger", func() {
+			endpoint := t.CreateLocalHostEndpoint()
+			_ = t.CreateEndpoint(t.newSubmEndpoint(healthCheckIP1))
+			t.pingerMap[healthCheckIP1].AwaitNoStart()
+
+			t.DeleteEndpoint(endpoint.Name)
+			t.pingerMap[healthCheckIP1].AwaitStart()
+		})
+	})
+})
+
+type testDriver struct {
+	*eventtesting.ControllerSupport
+	pingerMap map[string]*fake.Pinger
+	handler   event.Handler
+	endpoints dynamic.ResourceInterface
+	client    submarinerv1client.RouteAgentInterface
+	stopCh    chan struct{}
+}
+
+func newTestDriver(enableHealthCheck bool) *testDriver {
+	t := &testDriver{
+		ControllerSupport: eventtesting.NewControllerSupport(),
+	}
+
+	BeforeEach(func() {
+		t.stopCh = make(chan struct{})
+
+		clientset := fakeClient.NewSimpleClientset()
+
+		dynamicClient := dynamicfake.NewSimpleDynamicClient(kubeScheme.Scheme)
+
+		t.endpoints = dynamicClient.Resource(submarinerv1.SchemeGroupVersion.WithResource("endpoints")).Namespace(namespace)
+		t.client = clientset.SubmarinerV1().RouteAgents(namespace)
+		t.pingerMap = map[string]*fake.Pinger{
+			healthCheckIP1: fake.NewPinger(healthCheckIP1),
+		}
+
+		config := &healthchecker.Config{
+			PingInterval:             1, // Set interval to 1 second for faster testing
+			MaxPacketLossCount:       1,
+			HealthCheckerEnabled:     enableHealthCheck,
+			RouteAgentUpdateInterval: 100 * time.Millisecond,
+		}
+
+		config.NewPinger = func(pingerCfg pinger.Config) pinger.Interface {
+			defer GinkgoRecover()
+			Expect(pingerCfg.Interval).To(Equal(time.Second *
+				time.Duration(config.PingInterval))) //nolint:gosec // We can safely ignore integer conversion error
+			Expect(pingerCfg.MaxPacketLossCount).To(Equal(config.MaxPacketLossCount))
+
+			p, ok := t.pingerMap[pingerCfg.IP]
+			Expect(ok).To(BeTrue())
+
+			return p
+		}
```

Comment:
Instead of passing in `enableHealthCheck`  to `newTestDriver`, it's better to default  `config.HealthCheckerEnabled` to true and have spec(s) set it to false in their own `BeforeEach` and move L357-359 below to `JustBeforeEach`.

Reply:
Done.